tests/integration/protostar_fixture.py

```diff
@@ -213,6 +217,46 @@ def create_migration_file(
         )
         return file_path
 
+    async def run_test_case(self, test_case_content: str):
```

Comment:
I don't like this method in comparison with existing flow because of following reasons (in such degree that I am hesistant to approve it):
1. It templates too much of the source file, not allowing passing setup hooks, controlling builtins, parameters, names.
2. It's less performant because it prevents bundling multiple related test cases together. This technique amortizes times like compilation.

Reply:
Removed unnecessary integration test.

Reply:
Currently we were inspecting errors in integration tests by looking up for a `TestResult` in the `TestingSummary` returned from the fixture.

> This approach allows running specific test cases, which makes debugging easier.

This is a fair point about a feature that we are missing, but retrofitting it into existing method is easy.

> Such functionality can be added later.
> 
> `protostar.run_test_case("...", setup_fn="...", setup="...", params={"a": "felt"}, use_syscall_ptr=True)`

Doesn't this look like too much magic for you? I would definitely prefer to write full test in Cairo instead of relying on wild kwargs here.

Reply:
Maybe `summary` should be a module-scoped fixture?

Reply:
`summary` is a result of running the test itself. I don't see how this can be a fixture.

Reply:
I've just pushed the code with this approach.